Fix for Instagram

0bmay opened the pull request:
This PR fixes pulling Profile posts, Hashtag searches and Location searches.  Location only allows for 1 page of data for users that are not logged in, so it only attempts to get 1 page of data.  Profile and Hashtag work as before pulling until there is nothing left.

Adds getting Video URLs for video posts

0bmay committed:
Instagram is mostly working

Just need date from location searches

0bmay committed:
Location Data has date now

but can only query 1 page of data

0bmay committed:
Location only gets 1 page of data

tsia

0bmay committed:
add id to InstagramPost class

tsia

0bmay committed:
removed commented out code

tsia

TheTechRobo commented on snscrape/modules/instagram.py:
```diff
@@ -53,24 +55,28 @@ def __init__(self, **kwargs):
 		super().__init__(**kwargs)
 		self._headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36'}
 		self._initialPage = None
+		self._api_url = None
```
I think camelCase should be used for variables?

0bmay replied:
I believe the preferred python syntax is not camel case, but this repo does use it. [PEP-8](https://peps.python.org/pep-0008/#function-and-variable-names)

TheTechRobo commented on snscrape/modules/instagram.py:
```diff
@@ -53,24 +55,28 @@ def __init__(self, **kwargs):
-			  commentsDisabled = node['node']['comments_disabled'],
-			  isVideo = node['node']['is_video'],
-			 )
+				url=url,
```
I think a space before and a space after the equal signs

0bmay replied:
same comment as before.. python style in PEP-8 says parameters should not have spaces around the equal sign... but this are minor change an can be made, even if it breaks the python style (and I see more issues in my IDE)

TheTechRobo replied:
The code doesn't use PEP8, no. Cf. #958

TheTechRobo reviewed (commented):
Just two minor style nits

0bmay committed:
formatting changes to match repo style

tsia

0bmay committed:
Add medium to InstagramPost

add dataclasses Medium, Photo, Video and VideoVariant for Instagram as logged out users only get 1 media type per post this simplifies the class to use 1 property with different class values
